src/main/webapp/scripts/perspective.js

```diff
@@ -1,21 +1,30 @@
+/*
+ *  Makes a new comment object form the text box
+ *  If it is valid, sends a Request object to Perspective API and updates DOM
+ *  If not valid, updates DOM with an error message
+ */
+
 function getToxicity() {
-  API_KEY = '';
-  const commentText = document.getElementById("comment-text").value;
-  const commentObject = {
-    'comment': {'text': commentText},
-    'requestedAttributes': {'TOXICITY': {}},
-    'languages': ['en']
-  };
-  const json = JSON.stringify(commentObject);
+  const comment = new Comment(document.getElementById("comment-text").value);
+  if(!comment.isValid) {
+    updateToxicityDom(comment.getError(), 'toxicity-container');
+    return;
+  }
   const request = new Request(
     `https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=${API_KEY}`,
-    {method: 'POST', body: json}
+    {
+      method: 'POST',
+      body: comment.perspectiveString;
+    }
   );
-  fetch(request)
+  fetch(comment.perspectiveRequest)
     .then(response => response.json())
     .then((responseJson) => {
       const score = responseJson.attributeScores.TOXICITY.summaryScore.value;
-      const containerDOM = document.getElementById('toxicity-container');
-      containerDOM.innerText = (score*100).toFixed(2) + '% likely to be toxic.';
+      updateToxicityDom((score*100).toFixed(2), 'toxicity-container');
+    },
+    function(err) {
```

Comment:
nit: I would probably use catch(error)=>{...} here, because it explicitly shows this is a try-catch block, and also consistent with other arrow functions in getToxicity()